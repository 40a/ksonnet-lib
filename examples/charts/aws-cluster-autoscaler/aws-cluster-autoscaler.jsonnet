local chart = import "../chart.libsonnet";
local maintainer = chart.maintainer;

local core = import "../../../kube/core.libsonnet";
local container = core.v1.container;
local persistent = core.v1.volume.persistent;
local mount = core.v1.volume.mount;
local port = core.v1.port;
local service = core.v1.service;
local volume = core.v1.volume;

local kubeUtil = import "../../../kube/util.libsonnet";
local deployment = core.extensions.v1beta1.deployment + kubeUtil.app.v1beta1.deployment;

local values = import "./values.libsonnet";

local template = import "template.libsonnet";

{
  // Configuration.

  local chartSpec =
    chart.Default("aws-cluster-autoscaler", "0.2.1") +
    chart.Description("Scales worker nodes within autoscaling groups.") +
    chart.Source("https://github.com/kubernetes/contrib/tree/master/cluster-autoscaler/cloudprovider/aws") +
    chart.Maintainer(
      maintainer.Default("Michael Goodness", "mgoodness@gmail.com")),

  "chart.json": chartSpec,
  "service.json":
    chart.DefaultService(fullname, name, chartSpec, release) +
    service.Port(port.service.WithTarget(values.service.servicePort, 8085)) +
    service.Type(values.service.type) +
    service.Annotations(values.service.annotations) +
    service.ClusterIp(values.service.clusterIp) +
    service.ExternalIps(values.service.externalIps) +
    service.LoadBalancerIp(values.service.loadBalancerIp) +
    service.LoadBalancerSourceRanges(values.service.loadBalancerSourceRanges),
  "deployment.json":
    local certPath = "/etc/ssl/certs/ca-certificates.crt";
    local certsVolume = volume.hostPath.Default("ssl-certs", certPath);
    local appContainer =
      container.Default(name, containerImage, values.image.pullPolicy) +
      container.Command(containerCommand) +
      container.Env([{"AWS_REGION": values.awsRegion}]) +
      container.Port(port.container.Default(8085)) +
      container.VolumeMounts([mount.FromVolume(certsVolume, certPath, true)]) +
      container.Resources(values.resources);
    deployment.FromContainer(fullname, values.replicaCount, appContainer) +
    deployment.PodAnnotations(values.podAnnotations) +
    deployment.PodLabel("release", release.name) +
    deployment.NodeSelector(values.nodeSelector),

  // Data

  local containerImage = "%s:%s" % [values.image.repository, values.image.tag],

  local autoscaleFlags = [
    "--nodes=%s:%s:%s" % [group.minSize, group.maxSize, group.name]
    for group in values.autoscalingGroups
  ],

  local extraArgsFlags = [
    "--%s=%s" % [key, values.extraArgs[key]],
    for key in std.objectFields(values.extraArgs)
  ],
  local containerCommand =
    [
      "./cluster-autoscaler",
      "--cloud-provider=aws"
    ] +
    autoscaleFlags + [
      "--scale-down-delay=%s" % values.scaleDownDelay,
      "--skip-nodes-with-local-storage=%s" % values.skipNodes.withLocalStorage,
      "--skip-nodes-with-system-pods=%s" % values.skipNodes.withSystemPods,
      "--v=4"
    ] +
    extraArgsFlags,

  local name = chart.Name(chartSpec.name),
  local fullname = chart.Fullname(chartSpec.name, "beta"),

  local release = {
    name: std.extVar("release.name"),
    service: std.extVar("release.service"),
  },
}
