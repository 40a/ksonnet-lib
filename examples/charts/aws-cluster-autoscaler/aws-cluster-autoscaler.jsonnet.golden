{
   "chart.json": {
      "apiVersion": "v1",
      "description": "Scales worker nodes within autoscaling groups.",
      "engine": "gotpl",
      "maintainers": [
         {
            "email": "mgoodness@gmail.com",
            "name": "Michael Goodness"
         }
      ],
      "name": "aws-cluster-autoscaler",
      "sources": [
         "https://github.com/kubernetes/contrib/tree/master/cluster-autoscaler/cloudprovider/aws"
      ],
      "version": "0.2.1"
   },
   "deployment.json": {
      "apiVersion": "extensions/v1beta1",
      "kind": "Deployment",
      "metadata": {
         "labels": { },
         "name": "aws-cluster-autoscaler-beta"
      },
      "spec": {
         "nodeSelector": { },
         "replicas": 1,
         "template": {
            "metadata": {
               "annotations": { },
               "labels": {
                  "app": "aws-cluster-autoscaler",
                  "release": "3"
               }
            },
            "spec": {
               "containers": [
                  {
                     "command": [
                        "./cluster-autoscaler",
                        "--cloud-provider=aws",
                        "--scale-down-delay=10m",
                        "--skip-nodes-with-local-storage=false",
                        "--skip-nodes-with-system-pods=true",
                        "--v=4"
                     ],
                     "env": [
                        {
                           "AWS_REGION": "us-east-1"
                        }
                     ],
                     "image": "gcr.io/google_containers/cluster-autoscaler:v0.4.0",
                     "imagePullPolicy": "IfNotPresent",
                     "name": "aws-cluster-autoscaler",
                     "ports": [
                        {
                           "containerPort": 8085
                        }
                     ],
                     "resources": { },
                     "volumeMounts": [
                        {
                           "mountPath": "/etc/ssl/certs/ca-certificates.crt",
                           "name": "ssl-certs",
                           "readOnly": true
                        }
                     ]
                  }
               ],
               "volumes": [ ]
            }
         }
      }
   },
   "service.json": {
      "apiVersion": "v1",
      "kind": "Service",
      "metadata": {
         "annotations": { },
         "labels": {
            "app": "aws-cluster-autoscaler",
            "chart": "aws-cluster-autoscaler",
            "heritage": "tiller",
            "release": "3"
         },
         "name": "aws-cluster-autoscaler-beta",
         "namespace": "aws-cluster-autoscaler-beta"
      },
      "spec": {
         "clusterIP": "",
         "externalIPs": [ ],
         "loadBalancerIP": "",
         "loadBalancerSourceRanges": [ ],
         "ports": [
            {
               "port": 8085,
               "targetPort": 8085
            }
         ],
         "selector": {
            "app": "aws-cluster-autoscaler",
            "release": "3"
         },
         "type": "ClusterIP"
      }
   }
}
